### PR TITLE
Fix namespace failover behavior

### PR DESCRIPTION
**Pull request by wxing1292**
**What changed?**
* If all attributes are specified, fail the request, only allow either
  * Failover namespace
  * Update namespace config
* When failing over, also reset replication state to normal

**Why?**
Fixing behavior issue & allow CLI to reset replication state from handover to normal

**How did you test it?**
UT

**Potential risks**
N/A

**Is hotfix candidate?**
N/A

### CODE REVIEW

File: service/frontend/namespace_handler.go [hunk 1]
```diff
@@ -893,7 +903,7 @@ func (d *namespaceHandlerImpl) maybeUpdateFailoverHistory(
 	if l := len(namespaceDetail.ReplicationConfig.FailoverHistory); l > 0 {
 		lastFailoverVersion = namespaceDetail.ReplicationConfig.FailoverHistory[l-1].FailoverVersion
 	}
-	if lastFailoverVersion != newFailoverVersion {
```
Comment (yux0) [inline service/frontend/namespace_handler.go:896]: should we just check if the lastFailoverVersion != oldFailoverVersion?

File: service/frontend/namespace_handler.go [hunk 2]
```diff
@@ -585,7 +586,7 @@ func (d *namespaceHandlerImpl) UpdateNamespace(
 		}
 	}
 
-	if configurationChanged && activeClusterChanged && isGlobalNamespace {
```
Comment (yux0) [inline service/frontend/namespace_handler.go:588]: do we already block local namespace update active cluster?

File: service/frontend/namespace_handler.go [hunk 3]
```diff
@@ -561,6 +561,7 @@ func (d *namespaceHandlerImpl) UpdateNamespace(
 		if updateReplicationConfig.GetActiveClusterName() != "" {
 			activeClusterChanged = true
 			replicationConfig.ActiveClusterName = updateReplicationConfig.GetActiveClusterName()
+			replicationConfig.State = enumspb.REPLICATION_STATE_NORMAL
```
Comment (yux0) [inline service/frontend/namespace_handler.go:564]: you might also have this in namespace replication task executor

File: service/frontend/namespace_handler.go (hunk 2 above)
Comment (wxing1292) [reply service/frontend/namespace_handler.go:588]: yes, and added a UT to validate

File: service/frontend/namespace_handler.go (hunk 3 above)
Comment (wxing1292) [reply service/frontend/namespace_handler.go:564]: what do you mean?

File: service/frontend/namespace_handler.go (hunk 1 above)
Comment (wxing1292) [reply service/frontend/namespace_handler.go:896]: actually, the impl of namespace failover history is confusing
1. the history is not replicated
2. when will the `lastFailoverVersion == newFailoverVersion`? this is the original check?

Comment (meiliang86) [reply service/frontend/namespace_handler.go:896]: lastFailoverVersion should always equal to oldFailoverVersion?

File: service/frontend/namespace_handler.go (hunk 3 above)
Comment (meiliang86) [reply service/frontend/namespace_handler.go:564]: So if for whatever reason active cluster is not changed but this status is set, we cannot unset it?

Comment (wxing1292) [reply service/frontend/namespace_handler.go:564]: or we can only failover to unset the state var. 

i guess the lifecycle is not entire clear, let us talk

Comment (yux0) [reply service/frontend/namespace_handler.go:564]: I mean here: https://github.com/temporalio/temporal/blob/4b4dbf3978ed2df5e25792fec9ca5524a1647080/common/namespace/replication_task_executor.go#L296-L302. If the namespace is in handover state and the namespace is failed over from another cluster. Then the replication path should also reset the state.

**Review by yux0 — APPROVED** [review]

**Pull request closed without merging**